manager/src/main/java/com/redhat/service/bridge/manager/BridgesServiceImpl.java

```diff
@@ -81,11 +70,20 @@ protected Bridge doCreateBridge(String customerId, BridgeRequest bridgeRequest)
         bridge.setSubmittedAt(ZonedDateTime.now(ZoneOffset.UTC));
         bridge.setCustomerId(customerId);
         bridge.setShardId(shardService.getAssignedShardId(bridge.getId()));
-        bridgeDAO.persist(bridge);
+
+        doCreateBridge(bridge);
 
         return bridge;
     }
 
+    @Transactional
+    // Bridge and Work creation should always be in the same transaction
+    protected void doCreateBridge(Bridge bridge) {
```

Comment:
@manstis I personally don't see the value of us keeping the `doCreateBridge` any more. We can just hoist the calls here into `createBridge`, add `@Transactional` to `createBridge` and then delete this method IMHO.

Reply:
@robpblake I was being conscious to make the transactional code as short as possible to avoid optimistic lock contention. However since I guess a single record will only ever be updated by either the `manager` or `shard` we're unlikely to ever have contention as both operate on the same record at quite different times.

Reply:
I'll make the change next week.

Reply:
@robpblake Updates completed.